test/extended/openstack/loadbalancers.go

```diff
@@ -382,6 +383,152 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack][lb][Serial] The O
+		e2e.Logf("Found expected allowed_cidrs '%v' in Openstack LoadBalancer Listener '%s'", listeners[0].AllowedCIDRs, listenerId)
+
+		// Check there is no connectivity
+		g.By("Connectivity checks - there should be no connectivity")
+		var connNumber int = 5
+		g.By(fmt.Sprintf("Trying to access the service %d times from outside and storing the name of the pods answering", connNumber))
+		results := make(map[string]int)
+		for i := 0; i < connNumber; i++ {
+			// https://github.com/kubernetes/kubernetes/blob/master/test/images/agnhost/README.md#netexec
+			podName, err := getPodNameThroughUdpLb(svcIp, fmt.Sprintf("%d", svcPort), "hostname")
+			if err != nil {
+				e2e.Logf("No connectivity (as expected) while accessing the LoadBalancer service on try %d: %q", i, err)
+			} else {
+				results[podName]++
+			}
+		}
+		e2e.Logf("Pods accessed after %d UDP requests:\n%v\n", connNumber, results)
+		o.Expect(len(results)).Should(o.Equal(0), "There is connectivity to svc pods when it shouldn't: %s", results)
+		e2e.Logf("No connectivity to the service as expected")
+
+		// Remove the LoadBalancerSourceRanges spec from the service
+		_, err = jig.UpdateService(func(svc *v1.Service) {
+			svc.Spec.LoadBalancerSourceRanges = nil
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		e2e.Logf("Removed LoadBalancerSourceRanges spec from the service")
+
+		// Wait until allowed_cidrs in the LB listener is updated to 0.0.0.0/0 (all traffic allowed)
+		allowAllAllowedCidrs := []string{"0.0.0.0/0"}
+		e2e.Logf("Expected allowed_cidrs: '%v'", allowAllAllowedCidrs)
+		o.Eventually(func() []string {
+			lbListener, err := octavialisteners.Get(loadBalancerClient, listenerId).Extract()
+			if err != nil {
+				e2e.Logf("Error ocurred: %v, trying next iteration", err)
+				return []string{}
+			}
+			e2e.Logf("Found AllowedCIDRs: %v", lbListener.AllowedCIDRs)
+			return lbListener.AllowedCIDRs
+
+		}, "10s", "1s").Should(o.ConsistOf(allowAllAllowedCidrs), "Didn't find the expected allowed_cidrs '%v'", allowAllAllowedCidrs)
+		e2e.Logf("Found expected allowed_cidrs '%v' in Openstack LoadBalancer Listener '%s'", allowAllAllowedCidrs, listenerId)
+
+		// Wait until ProvisioningStatus in the LB listener is back to ACTIVE after it's been updated
+		activeProvisioningStatus := "ACTIVE"
+		e2e.Logf("Expected ProvisioningStatus: '%s'", activeProvisioningStatus)
+		o.Eventually(func() string {
+			lbListener, err := octavialisteners.Get(loadBalancerClient, listenerId).Extract()
+			if err != nil {
+				e2e.Logf("Error ocurred: %v, trying next iteration", err)
+				return ""
+			}
+			e2e.Logf("Found ProvisioningStatus: %s", lbListener.ProvisioningStatus)
+			return lbListener.ProvisioningStatus
+
+		}, "10s", "1s").Should(o.Equal(activeProvisioningStatus), "Didn't find the expected ProvisioningStatus '%s'", activeProvisioningStatus)
```

Comment:
Just 10 seconds of waiting? This feels like something that could fail often with Amphora.

Reply:
I added an active wait because it's not immediate, but it takes around 1 sec in most cases (with amphora).
```
Mar 10 04:25:10.729: INFO: Expected allowed_cidrs: '[0.0.0.0/0]'
Mar 10 04:25:10.792: INFO: Found AllowedCIDRs: [9.9.9.9/32]
Mar 10 04:25:11.858: INFO: Found AllowedCIDRs: [0.0.0.0/0]
Mar 10 04:25:11.858: INFO: Found expected allowed_cidrs '[0.0.0.0/0]' in Openstack LoadBalancer Listener '75935039-b393-4349-acb5-1fa3adfb86f3'
```

Reply:
But that's a log of API value check above, not "Found expected ProvisioningStatus". I'm afraid it might take longer for Amphora to actually process this sometimes. I couldn't find this log in the gate.

Reply:
My fault... I pasted the incorrect log lines, these are the correct logs:

```
Mar 10 11:44:52.185: INFO: Expected ProvisioningStatus: 'ACTIVE'
Mar 10 11:44:52.344: INFO: Found ProvisioningStatus: PENDING_UPDATE
Mar 10 11:44:53.512: INFO: Found ProvisioningStatus: PENDING_UPDATE
Mar 10 11:44:54.665: INFO: Found ProvisioningStatus: PENDING_UPDATE
Mar 10 11:44:56.105: INFO: Found ProvisioningStatus: ACTIVE
Mar 10 11:44:56.105: INFO: Found expected ProvisioningStatus 'ACTIVE' in Openstack LoadBalancer Listener '539337ad-8fcf-4ba1-9165-ccd495c6e6d9'
```

The log is from the test that failed in the gate: https://prow.ci.openshift.org/61bfa262-82c4-4dc5-86f9-9006a1e609df
and it took less than 4 seconds to update the listener for an Amphora.

Do you think that increasing the timeout, to 20s for instance, is reasonable?

Reply:
Yes, I think this strongly depends on how stressed the env is and may take longer than that. Let's maybe put 30 seconds in there just to be safe here from false positives.

Reply:
Done!